nucypher/characters/lawful.py

```diff
@@ -1388,3 +1345,118 @@ def encrypt_message():
+        def __contains__(self, checksum_address: str) -> bool:
+            return bool(checksum_address in self.__accounts)
+
+        @property
+        def active_account(self) -> str:
+            return self.blockchain.transacting_power.account
+
+        def __get_accounts(self) -> None:
+            accounts = self.blockchain.client.accounts
+            self.__accounts.update(accounts)
+
+        @property
+        def accounts(self) -> set:
+            return self.__accounts
+
+        @validate_checksum_address
+        def activate_account(self, checksum_address: str, password: str = None) -> None:
+            if checksum_address not in self:
+                self.__get_accounts()
+                if checksum_address not in self:
+                    raise self.UnknownAccount
+            try:
+                transacting_power = self.__transacting_powers[checksum_address]
+            except KeyError:
+                transacting_power = TransactingPower(password=password, account=checksum_address)
+                self.__transacting_powers[checksum_address] = transacting_power
+            transacting_power.activate(password=password)
+
+        @property
+        def balances(self) -> Dict[str, int]:
+            balances = dict()
+            for account in self.__accounts:
+                funds = {'ETH': self.blockchain.client.get_balance(account),  # TODO: EthAgent or something?
+                         'NU': self.token_agent.get_balance(account)}
+                balances.update({account: funds})
+            return balances
+
+    #
+    # StakeHolder
+    #
+
+    def __init__(self,
+                 is_me: bool = True,
+                 initial_address: str = None,
+                 checksum_addresses: set = None,
+                 password: str = None,
+                 *args, **kwargs):
+        super().__init__(is_me=is_me, checksum_address=initial_address, *args, **kwargs)
+        self.log = Logger(f"stakeholder")
+
+        # Wallet
+        self.wallet = self.StakingWallet(registry=self.registry, checksum_addresses=checksum_addresses)
+        if initial_address:
+            # If an initial address was passed,
+            # it is safe to understand that it has already been used at a higher level.
+            if initial_address not in self.wallet:
+                raise self.StakingWallet.UnknownAccount
+            self.assimilate(checksum_address=initial_address, password=password)
+
+    @validate_checksum_address
+    def assimilate(self, checksum_address: str, password: str = None) -> None:
+        self.wallet.activate_account(checksum_address=checksum_address, password=password)
+        original_form = self.checksum_address
+        self.checksum_address = checksum_address
+        self.stakes = StakeList(registry=self.registry, checksum_address=checksum_address)
+        self.stakes.refresh()
+        self.log.info(f"Resistance is futile - Assimilating Staker {original_form} -> {checksum_address}.")
```

Comment:
So `assimilate()` changes the "current" staker account to `checksum_address`. Is it correct? If so, I don't think the logged message is clear. Perhaps something like:
```suggestion
        self.log.info(f"Resistance is futile - Assimilating Staker {checksum_address} (previously it was {original_form})")
```